docs/DesignDocs/AttestationEvidenceFormatSupport.md

```diff
@@ -0,0 +1,118 @@
+Attestation Evidence Format Support
+=====
+
+One of the goal of OE SDK attestation design is to be TEE-agnostic. At the same time, OE SDK should enable a verifier application to verify evidences from different sources and TEEs, if the format is known to the verifier.
+
+Motivation
+----------
+
+Current evidence verify API `oe_verify_evidence()` verifies the evidences generated by `oe_get_evidence()` and returns the claims contained in the evidence. Such evidence has a specific format including an `oe_attestation_header`.
+As reported in issue #2999, attestation service has to verify evidences from different sources other than those generated by `oe_get_evidence()`. Such as evidence generated by legacy OE SDK API `oe_get_report()` and raw TEE evidence generation such as SGX quote generation mechanism. And like `oe_verify_evidence()` returns claims, attestation service is also required to return the supplemental information. To achieve that, a new verify API is needed to verify different evidence structures and return the required supplemental information.
+
+Goals:
+- Verifier can call OE SDK API to verify known evidence formats generated by either OE SDK API or other attesters.
+
+User Experience
+---------------
+
+New parameter `format_id` will be added to `oe_verify_evidence` to verify various evidences other than the ones generated by `oe_get_evidence()`, such as a raw SGX quote. To verify an evidence generated by `oe_get_evidence()`, `format_id` should be `null`.
```

Comment:
Is there a specific reason why the value is `null` for evidence created by `oe_get_evidence()`? It would seem more regular to assign that with its own (named) format_id, and to require format_id every time.

Reply:
I think that when the evidence data is from `oe_get_evidence()`, the `format_id` parameter is optional: can be NULL or a valid format ID (which must match the one in the evidence data).

Reply:
> Is there a specific reason why the value is `null` for evidence created by `oe_get_evidence()`? It would seem more regular to assign that with its own (named) format_id, and to require format_id every time.

Because when verify API reads a `null` format_id, then it knows the evidence buffer is an evidence generated by `oe_get_evidence()` and will unpack the evidence first (remove oe_attestation header and extract the `format_id` in the header). Otherwise, when verify API reads a format_id, there is no solid way to tell whether the evidence buffer is a raw evidence, or an evidence generated by `oe_get_evidence`.

Reply:
@qiucwang thank you, that makes complete sense, I'd misunderstood what happened in that case.